### PR TITLE
Rework the package to fit the new apps implementation

### PR DESCRIPTION
**Pull request by tamorim**

### CODE REVIEW

File: src/RegistryClient.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  RegistryVendor(account, workspace, vendor) {
+    return `${this.Registry(account, workspace)}/${vendor}/apps`;
+  },
+
+  RegistryApp(account, workspace, vendor, name) {
+    return `${this.RegistryVendor(account, workspace, vendor)}/${name}`;
+  },
+
+  RegistryAppVersion(account, workspace, vendor, name, version) {
+    return `${this.RegistryApp(account, workspace, vendor, name)}/${version}`;
```
Comment (brenoc) [inline src/RegistryClient.js]: Nice thing here :+1:

Comment (brenoc) [inline src/RegistryClient.js]: I mean, the composition

File: src/AppsClient.js [hunk 2]
```diff
@@ -20,105 +20,127 @@ class AppsClient {
...
+        context
+      }
     });
   }
 
-  getSettingsSchema(app, version) {
-    checkRequiredParameters({app, version});
-    const url = `${this.endpointUrl}${this.route.SettingsSchema(app, version)}`;
+  getApp(account, workspace, vendor, name, version, context = '') {
+    checkRequiredParameters({});
```
Comment (brenoc) [inline src/AppsClient.js]: Missing parameters

Comment (tamorim) [inline src/AppsClient.js]: Nice catch!

File: src/RegistryClient.js (hunk 1 above)
Comment (tamorim) [inline src/RegistryClient.js]: Thanks! :smiley:

File: src/AppsClient.js [hunk 3]
```diff
@@ -20,105 +20,127 @@ class AppsClient {
...
+  installApp(account, workspace, descriptor, ttl = null) {
+    checkRequiredParameters({account, workspace, descriptor});
+    const url = `${this.endpointUrl}${this.routes.Apps(account, workspace)}`;
+    const queryString = ttl ? {preReleaseTTL: ttl} : {};
 
-    return request.get({
+    return request.post({
       ...this.defaultRequestOptions,
       url,
+      qs: queryString,
```
Comment (firstdoit) [inline src/AppsClient.js]: You could avoid the colon by naming your const `qs`.

File: src/AppsClient.js [hunk 4]
```diff
@@ -20,105 +20,127 @@ class AppsClient {
...
-    const url = `${this.endpointUrl}${this.route.SettingsSchema(app, version)}`;
+  getApp(account, workspace, vendor, name, version, context = '') {
+    checkRequiredParameters({account, workspace, vendor, name, version});
+    const url = `${this.endpointUrl}${this.routes.App(account, workspace, vendor, name, version)}`;
 
     return request.get({
       ...this.defaultRequestOptions,
       url,
+      qs: {
+        context
```
Comment (firstdoit) [inline src/AppsClient.js]: Will always send `?context=` empty string?

File: src/AppsClient.js [hunk 5]
```diff
@@ -20,105 +20,127 @@ class AppsClient {
...
-  RootFolders(appId, version) {
-    return `/${appId.Vendor}/apps/${appId.Name}/${version}/files`;
+  Apps(account, workspace) {
+    return `/${account}/${workspace}/apps`;
   },
 
-  Files(appId, version, service) {
-    return `/${appId.Vendor}/apps/${appId.Name}/${version}/files/${service}`;
+  App(account, workspace, vendor, name, version) {
+    return `/${account}/${workspace}/apps/${vendor}.${name}@${version}`;
```
Comment (firstdoit) [inline src/AppsClient.js]: We need to extract these locator parsers and serializers. Also repeated here: https://github.com/vtex/render/blob/feature/boilerplate/src/resources/components.js#L8

File: src/RegistryClient.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
+import request from 'request-promise';
+import getEndpointUrl from './utils/appsEndpoints.js';
+import checkRequiredParameters from './utils/required.js';
+
+class RegistryClient {
```
Comment (firstdoit) [inline src/RegistryClient.js]: I would love for our team to discuss how to move away from stateful OO clients. I'm not sure what's the best pattern to replace it, though.

**Comment by firstdoit** [comment]
Some small questions. Overall, LGTM! 👍

File: src/AppsClient.js (hunk 3 above)
Comment (tamorim) [inline src/AppsClient.js]: @firstdoit I find `qs` a really bad name that carries some cognitive burden to be recognized, so I preferred to use `queryString`

File: src/AppsClient.js (hunk 5 above)
Comment (tamorim) [inline src/AppsClient.js]: It would be nice.

File: src/RegistryClient.js (hunk 6 above)
Comment (tamorim) [inline src/RegistryClient.js]: Neither do I, we need to search to see how other people are doing.

File: src/AppsClient.js (hunk 4 above)
Comment (tamorim) [inline src/AppsClient.js]: @firstdoit Yep!

File: src/AppsClient.js (hunk 5 above)
Comment (brenoc) [inline src/AppsClient.js]: That could be useful.

File: src/RegistryClient.js (hunk 6 above)
Comment (brenoc) [inline src/RegistryClient.js]: I think, the only way we could avoid this, is passing all the default parameters as a parameter. Eg:

```js
const clientConfig = AppClient.config({
  authToken,
  userAgent,
  endpointUrl
})

AppClient.publishApp(clientConfig, account, workspace, zip)
```

**Pull request merged**